monitor.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"time"
+)
+
+// Monitor is an interface implemented by a value that can record metrics about
+// the daemon.
+/*
+type Monitor interface {
+	// RecordServingTime records the amount of time it took to completely
+	// handle a request to the system (acceptance to full response).
+	RecordServingTime(time.Duration)
+	// IncrementAssignment increases the count of number of times an
+	// assignment was given a specific location.
+	IncrementLocation(string)
+	// IncrementErrorAssignment increases the count of the number of errors
+	// that occurred while processing an incoming assignment.
+	IncrementErrorHandlerAssignment()
+	// IncrementErrorRollout increases the count of the number of errors
+	// that occurred while processing a rollout value.
+	IncrementRollout(error)
+}
+*/
+type Monitor interface {
+	RecordServe(error)
+	RecordServingTime(time.Duration)
+	RecordHandling(string, error)
+	RecordRolloutUpdate(error)
+}
+
+// A DogStatsDMonitor represents a proxy for sending metrics to Datadog using
+// the namespace prefix "maxwellsdaemon.".
+type DogStatsDMonitor struct {
+	conn net.Conn
+}
+
+// NewDogStatsDMonitor creates a connection to the dogstatsd agent and prepares
+// to accept metrics.
+func NewDogStatsDMonitor() *DogStatsDMonitor {
+	return &DogStatsDMonitor{
+		conn: nil,
+	}
+}
+
+func (statsdMonitor *DogStatsDMonitor) send(s string) {
+	if statsdMonitor.conn == nil {
+		conn, err := net.Dial("udp", "127.0.0.1:8125")
```

Comment:
Should we inject this DD end point config, or it it guaranteed the DD agent is always on local box on port 8125?